search/django/weedid/models.py

```diff
@@ -1,3 +1,43 @@
-# from django.db import models
+from django.db import models
+from django.contrib.postgres.fields import JSONField
+from django.contrib.auth.models import AbstractUser
 
-# Create your models here.
+
+class Dataset(models.Model):
+    UPLOAD_STATUS_CHOICES = [
+        ("N", "None"),
+        ("P", "Processing"),
+        ("I", "Incomplete"),
+        ("AR", "Awaiting Review"),
+        ("F", "Failed"),
+        ("C", "Complete"),
+    ]
+    metadata = JSONField(null=True)
+    upload_id = models.CharField(
+        primary_key=True, max_length=20, null=False, blank=False, unique=True
+    )
+    upload_agcontext = JSONField(null=True)
```

Comment:
why is every field (except for `metadata`) prefixed with `upload_`? It seems redundant to me.

Reply:
I think we may also want to store a summary of some of the information found in weedcoco to show on the datasets page (although perhaps we can just as easily get this from elastic): which categories are present, and which ML tasks are supported.